packages/core/testing/src/test_bed_common.ts

```diff
@@ -41,8 +42,29 @@ export type TestModuleMetadata = {
   imports?: any[],
   schemas?: Array<SchemaMetadata|any[]>,
   aotSummaries?: () => any[],
+  teardown?: ModuleTeardownOptions;
 };
 
+/**
+ * @publicApi
+ */
+export interface TestEnvironmentOptions {
+  aotSummaries?: () => any[];
+  teardown?: ModuleTeardownOptions;
+}
+
+/**
+ * Object used to configure the test module teardown behavior in `TestBed`.
```

Comment:
I think we'd need to add more info here to explain why these options exist and provide recommendations on when they should be used/changed and what the plan is (that these options will eventually be deprecated and removed, so we do not encourage using them unless it's absolutely needed and blocking tests).

Reply:
The idea was to have a silent rollout of this in 12.1.0 and then add a page in the docs similar to https://angular.io/guide/migration-update-libraries-tslib.

Reply:
Thanks for additional info, I'd propose adding a couple sentences to describe that:
```suggestion
 * Object used to configure the test module teardown behavior in `TestBed`.
 * This option is exposed as a public API to allow switching to the legacy TestBed behavior temporarily
 * (where testing module was not removed as a part of the test teardown process) in case your tests rely on
 * this old behavior. These config options will be deprecated and removed in the future versions of Angular.
```

Reply:
I'm not sure that people who aren't familiar with the `TestBed` internals would know what this is referring to.

Reply:
We can update the message to make it more user-friendly. I'd like to include some information about the fact that these config options are temporary and developers should not rely on them.

Reply:
IMO we should capture the fact that it's temporary when we eventually add the page on AIO. Also the plan is to provide schematics that add the parameter and then remove it once we've changed the default value so hopefully people won't have to deal with it themselves.

Reply:
The `ModuleTeardownOptions` would become discoverable on the AIO once this PR is merged, see
https://pr42566-893884a.ngbuilds.io/api/core/testing/ModuleTeardownOptions.

Reply:
I meant that providing all the context behind the option would take a few paragraphs of text which seems like overkill for an API doc. E.g. the design doc for these changes takes 170 words to explain the issues.